Tests/Functional/OldModel/AbstractTimeSpanTest.php

```diff
@@ -111,4 +146,260 @@ public function getTimeForBeginTimeAndEndTimeOnDifferentDaysReturnsBothTimesWith
+
+    /**
+     * @test
+     */
+    public function getTimeForNoTimeNotCallsHook()
+    {
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::never())->method('modifyTimeSpan');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TimeSpan::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->getTime();
+    }
+
+    /**
+     * @test
+     */
+    public function getTimeForBeginTimeOnlyNotCallsHook()
+    {
+        $this->subject->setBeginDateAndTime(\mktime(9, 50, 0, 1, 1, 2010));
+
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::never())->method('modifyTimeSpan');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TimeSpan::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->getTime();
+    }
+
+    /**
+     * @test
+     */
+    public function getTimeForEndTimeOnlyNotCallsHook()
+    {
+        $this->subject->setEndDateAndTime(\mktime(18, 30, 0, 1, 1, 2010));
+
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::never())->method('modifyTimeSpan');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TimeSpan::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->getTime();
+    }
+
+    /**
+     * @test
+     */
+    public function getTimeForBeginTimeAndEndTimeOnSameTimeNotCallsHook()
+    {
+        $this->subject->setBeginDateAndTime(\mktime(9, 50, 0, 1, 1, 2010));
+        $this->subject->setEndDateAndTime(\mktime(9, 50, 0, 1, 3, 2010));
+
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::never())->method('modifyTimeSpan');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TimeSpan::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->getTime();
+    }
+
+    /**
+     * @test
+     */
+    public function getTimeForBeginTimeAndEndTimeOnDifferentTimesCallsHookAndReturnsModifiedValue()
+    {
+        $this->subject->setBeginDateAndTime(\mktime(9, 50, 0, 1, 1, 2010));
+        $this->subject->setEndDateAndTime(\mktime(18, 30, 0, 1, 1, 2010));
+
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::once())->method('modifyTimeSpan')
+            ->with('09:50&#8211;18:30', '09:50', '&#8211;', '18:30')
+            ->willReturn('modified');
```

Comment:
I propose putting `modified` in a variable to make the connection more obvious. Ditto in the other respective test.